### DISCUSSION TITLE
Next.js + styled-components

### DISCUSSION THREAD

**Discussion by rhuangabrielsantos** [frontendbr/forum#2011] [Perguntas e Respostas] [answered]
Oi pessoal, estou com uma duvida sobre um problema que está acontecendo comigo.

Criei um site simples chamado **[Github Profile](github-profile-lovat.vercel.app)**, e ao acessar a pagina o css não está carregado ainda e é possível ver por alguns segundos o site sem estilização. O que pode estar acontecendo e como faço para resolver?

Link do repositório: https://github.com/rhuangabrielsantos/GithubProfile

Muito obrigado!

**Comment by giovannism20** [comment]
@rhuangabrielsantos Testei aqui e está aparecendo normalmente...

**Comment by vitebo** [comment]
você ta fazendo o build do conteúdo para mandar para o servidor?
eu chuto que pode ser isso por não ter encontrado nenhum script de `build` no seu `package.json`

**Reply by rhuangabrielsantos** [reply]
O Build está sendo feito diretamente pelas configurações da vercel, o comentário do @hmhallak solucionou o meu problema, mas obrigado! :)

**Comment by hmhallak** [comment, marked as answer]
Dá uma olhada aqui na documentação do styled-components: https://styled-components.com/docs/advanced#nextjs

> Next.js
Basically you need to add a custom pages/_document.js (if you don't have one). Then copy the logic for styled-components to inject the server side rendered styles into the `<head>`.
Refer to our example in the Next.js repo for an up-to-date usage example.

Vc só precisa adicionar um arquivo em pages/_document.js com o código de exemplo para habilitar o server side rendering com o styled-components.

Link do arquivo de exemplo: https://github.com/vercel/next.js/blob/master/examples/with-styled-components/pages/_document.js

**Reply by rhuangabrielsantos** [reply]
Muito obrigado @hmhallak !!

É exatamente isso!
